### PR TITLE
Add PkgEqual and HashEqual predictaes -- testing + code planning

### PR DESCRIPTION
**Pull request by rmetzman**
# Description of the PR

Testing + code planning for PkgEqual and HashEqual predictaes

# PR Checklist

- [x] All commits have [a Developer Certificate of Origin (DCO)](https://wiki.linuxfoundation.org/dco) -- they are generated using `-s` flag to `git commit`.
- [x] All new changes are covered by tests
- [x] If GraphQL schema is changed, `make generate` has been run
- [x] If `collectsub` protobuf has been changed, `make proto` has been run
- [x] All CI checks are passing (tests and formatting)
- [x] All dependent PRs have already been merged

### CODE REVIEW

File: pkg/guacanalytics/patchPlanning.go [hunk 1]
```diff
@@ -230,6 +234,18 @@ func exploreHasSourceAtFromSource(ctx context.Context, gqlClient graphql.Client,
 	return nil
 }
 
+// TODO: Expand to not just deal with packageVersions
+func explorePkgEqual(ctx context.Context, gqlClient graphql.Client, q *queueValues, pkgEqual model.NeighborsNeighborsPkgEqual) {
+	// Step 1: Loop through all the packages listed in the pkgEqual
+	// Step 2: as long as they are not equal to the current node, add them and their packageName to the queue
```
Comment (lumjjb) [inline pkg/guacanalytics/patchPlanning.go:240]: do we need to add packageVersion to queue as well?

**Review by lumjjb — COMMENTED** [review]
psuedocode lgtm, minor comments

File: pkg/guacanalytics/patchPlanning.go (hunk 1 above)
Comment (rmetzman) [reply pkg/guacanalytics/patchPlanning.go:240]: Yes, good point.  Will change that.

File: pkg/guacanalytics/patchPlanning_test.go [hunk 2]
```diff
@@ -784,6 +917,52 @@ func ingestCertifyGood(ctx context.Context, client graphql.Client, graph assembl
...
+		if err != nil {
+			return fmt.Errorf("error in ingesting PkgEqual: %v\n", err)
+		}
+	}
+	return nil
+}
+
+func ingestHashEqual(ctx context.Context, client graphql.Client, graph assembler.IngestPredicates) error {
+	for _, ingest := range graph.HashEqual {
+		_, err := model.IngestArtifact(context.Background(), client, *ingest.Artifact)
```
Comment (pxp928) [inline pkg/guacanalytics/patchPlanning_test.go:945]: use the pass in `ctx` here instead of `context.Background()` for all these.

**Review by pxp928 — APPROVED** [review]
LGTM, minor nit.

**Pull request merged**